src/application.coffee

```diff
@@ -217,9 +218,27 @@ fetch = (app, setDeviceUpdateState = true) ->
 
 			if conf['RESIN_SUPERVISOR_DELTA'] == '1'
 				logSystemEvent(logTypes.downloadAppDelta, app)
-				requestTimeout = checkInt(conf['RESIN_SUPERVISOR_DELTA_REQUEST_TIMEOUT'], positive: true) ? 30 * 60 * 1000
-				totalTimeout = checkInt(conf['RESIN_SUPERVISOR_DELTA_TOTAL_TIMEOUT'], positive: true) ? 24 * 60 * 60 * 1000
-				dockerUtils.rsyncImageWithProgress(app.imageId, { requestTimeout, totalTimeout }, onProgress)
+
+				Promise.join utils.getConfig('apiKey'), utils.getConfig('uuid'), (apiKey, uuid) ->
+					appImageInfos = app.imageId.split('/')
+					registryService = appImageInfos[0]
+					repositoryPath = appImageInfos.slice(1).join('/')
+					tokenEndpoint = "#{config.apiEndpoint}/registry/token" #TODO read from state endpoint
```

Comment:
Heads up: the endpoint has been changed to `/auth/v1/token` but in any case let's please sync on this before merging this PR